File: python/flyql/matcher/key.py
from typing import List
from flyql.core.key import parse_key


class Key:
    def __init__(
        self,
        value: str,
    ) -> None:
        parsed_key = parse_key(value)
        self.value = parsed_key.segments[0] if parsed_key.segments else value
        self.path: List[str] = (
            parsed_key.segments[1:] if len(parsed_key.segments) > 1 else []
        )

File: golang/matcher/key.go
package matcher

import "github.com/iamtelescope/flyql/golang"

type Key struct {
	Value string
	Path  []string
}

func NewKey(value string) Key {
	parsed, err := flyql.ParseKey(value, 0)
	if err != nil {
		return Key{
			Value: value,
			Path:  []string{},
		}
	}

	if len(parsed.Segments) == 0 {
		return Key{
			Value: value,
			Path:  []string{},
		}
	}

	return Key{
		Value: parsed.Segments[0],
		Path:  parsed.Segments[1:],
	}
}
